### PR TITLE
glkernel stages

### PR DESCRIPTION
**Pull request by Tobias1595**

### CODE REVIEW

File: source/gloperate-glkernel/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+target_link_libraries(${target}
+    PRIVATE
+    cppassist::cppassist
+    cppexpose::cppexpose
+    cpplocate::cpplocate
+    glkernel::glkernel
+    globjects::globjects
+    glbinding::glbinding
+    libzeug::signalzeug
```
Comment (scheibel) [inline source/gloperate-glkernel/CMakeLists.txt]: This one is obsolete and has to be removed.

File: source/gloperate-glkernel/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+#
+# Compile definitions
+#
+
+target_compile_definitions(${target}
+    PRIVATE
+
+    PUBLIC
+    $<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:${target_upper}_STATIC_DEFINE>
```
Comment (scheibel) [inline source/gloperate-glkernel/CMakeLists.txt]: We use ```${target_id}``` now, please refer to https://github.com/cginternals/cmake-init/blob/master/source/baselib/CMakeLists.txt#L150.

File: source/gloperate-glkernel/source/DiscDistributionKernelStage.cpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+
+namespace
+{
+
+
+using namespace glm;
+
+// push the corners of the square in to form a disc
+vec2 pushCorners(vec2 v)
```
Comment (scheibel) [inline source/gloperate-glkernel/source/DiscDistributionKernelStage.cpp]: Take a ```const vec2 & v``` as parameter.

File: source/gloperate-glkernel/source/TransparencyKernelStage.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+#include <algorithm>
+
+#include <glbinding/gl/enum.h>
+
+#include <globjects/Texture.h>
+
+
+namespace
+{
+    const size_t s_alphaValues = 256;
```
Comment (scheibel) [inline source/gloperate-glkernel/source/TransparencyKernelStage.cpp]: Those kernel sizes are configurable for the other stages.
Is there a particular reason the kernel sizes are constants here?

**Comment by scheibel** [comment]
Can you provide a summary of stages that are deterministic and the ones that use one form of (pseudo-) randomness to compute the kernels?

**Comment by Tobias1595** [comment]
Disc & hemisphere distribution kernels are fully deterministic.
Noise & transparency kernels use the C++ standard library's pseudo-random implementations.

**Pull request merged**